roles/firewalld_service_verify/tasks/main.yml

```diff
@@ -0,0 +1,57 @@
+---
+# vim: set ft=ansible:
+#
+# This role confirms that the 'firewalld' package/service is correctly
+# installed (or not installed) on the host under test.
+#
+- name: Default to firewalld being uninstalled
+  set_fact:
+    firewalld_installed: false
+
+- name: Determine if firewalld should be installed
+  when: (ansible_distribution == 'Fedora' and
+         ansible_distribution_major_version | version_compare('26', '>')) or
+        (ansible_distribution == 'RedHat' and
+         ansible_distribution_version | version_compare('7.4', '>'))
+  set_fact:
+    firewalld_installed: true
+
+- when: firewalld_installed
+  block:
+    - name: Verify firewalld package is installed
+      command: rpm -q firewalld
+
+    - name: Verify firewalld is disabled by default
+      command: systemctl is-enabled -q firewalld
+      register: is_enabled
+      failed_when: is_enabled.rc == 0
```

Comment:
Discussion point.  Do you think its clearer to look at return codes or use jinja2 filtering (`is_enabled|success`).  The use seems mixed in this role and I'm just wondering if we should choose one or the other for consistency and clarity if it provides the same logic.

Reply:
Good point.  I used `enable_start|success` because the `service` task doesn't have an `rc` attribute.  I think I prefer the Jinja2 filter approach overall.

I'll make changes to standardize on that.